### ISSUE TITLE
Third party cookies being allowed on www.figma.com

### ISSUE THREAD

**Issue by wongbryan** [GoogleChromeLabs/privacy-sandbox-dev-support#252]
I've disabled third party cookies by enabling the experiment flag and also via Setting > Tracking Protection > Block all third party cookies

If I go to www.figma.com and open the issue panel in Chrome, I see a message saying Figma is allowed to read third party cookies.
<img width="872" alt="Screenshot 2024-02-08 at 2 26 26 PM" src="https://github.com/GoogleChromeLabs/privacy-sandbox-dev-support/assets/26299742/82ba97dd-aaff-4c7e-801a-8dfb1cf048bf">

This does not happen on other websites, only on Figma. can anyone else repro?

**Comment by wanderview** [comment]
The movie does not show the figma tab being reloaded after changing the cookie setting.  Did you try that?

Also, can you please show your chrome://flags page?

**Comment by wongbryan** [comment]
I restarted chrome multiple times. Unable to send screenshot of
chrome://flags atm, but I have enabled the third party cookies deprecation
flag and that’s it

**Comment by wongbryan** [comment]
here is a SS of my chrome://flags page 
<img width="835" alt="image" src="https://github.com/GoogleChromeLabs/privacy-sandbox-dev-support/assets/26299742/b14f4834-b458-4796-ad67-a4083a50d014">

**Comment by wanderview** [comment]
Have you added a cookie exception by clicking this icon in the omnibox?

![image](https://github.com/GoogleChromeLabs/privacy-sandbox-dev-support/assets/1256526/1f4d6def-3ade-447e-a996-0de38ce6fe40)

You would then see something in chrome://settings/trackingProtection like this: 

![image](https://github.com/GoogleChromeLabs/privacy-sandbox-dev-support/assets/1256526/bca9cf8d-1a3c-49d0-abfd-b35dbb627f2e)

**Comment by samdutton** [comment]
If you block all third-party cookies from `chrome://settings/trackingProtection`, third-party cookies should be blocked for figma.com even if it's in the "Sites allowed to use third-party cookies" list (as per the attached screenshot). Not sure why that's not happening for you.

Just to double check: have you tried hard-refreshing figma.com after changing `chrome://settings/trackingProtection` to block all third-party cookies?

Not sure if it's relevant, but are you in a Chrome Enterprise environment?

![image](https://github.com/GoogleChromeLabs/privacy-sandbox-dev-support/assets/205226/0ccffd1b-dd4e-4949-8a41-988a1e72db90)
![image](https://github.com/GoogleChromeLabs/privacy-sandbox-dev-support/assets/205226/86de18c1-8caa-4519-83c7-a3a33adfa858)

**Comment by wongbryan** [comment]
@wanderview I have not added a cookie exception for Figma. See attached screenshot (the omnibox says third party cookies are blocked, but the issues panel disagrees):

<img width="1720" alt="image" src="https://github.com/GoogleChromeLabs/privacy-sandbox-dev-support/assets/26299742/86b7311b-83b1-43db-896f-06e679e091ab">

@samdutton I have hard refreshed and the issues panel has the same message. I am in a chrome enterprise environment (on a work laptop)

**Comment by wanderview** [comment]
It seems possible your enterprise admin is applying a cookie exception via an enterprise admin.

**Comment by wanderview** [comment]
You can observe enterprise policies on chrome://policy.  Look for an entry with the policy name "CookiesAllowedForUrls".

**Comment by wongbryan** [comment]
@wanderview There is no entry for "CookiesAllowedForUrls" and figma.com is not listed in any of the policy domains. I just found this page: https://github.com/amaliev/3pcd-exemption-heuristics/blob/main/explainer.md and though this is related, but I disabled the heuristics exceptions by disabling the Third-party Cookie Grants Heuristics Testing flag, Figma is still able to read third party cookies. I wonder though if there's some behavior in Figma's web editor that is causing 3PC exceptions. Again, can you verify that you are unable to repro this? Please note that this issue **only occurs in Figma's web editor UI, but not on the Figma homepage. You can repro by visiting this link and opening the chrome issues console while on this page:**

https://www.figma.com/file/rL9D4IbnGdiEZRhTkYRlE8/Untitled?type=design&mode=design&t=Gij6XOR8Avzzk5uY-1

**Comment by amaliev** [comment]
Hey @wongbryan , thanks for reporting this issue - it looks like the description in the Issues panel is poorly worded. This message is shown on sites that have 3PCD mitigations enabled, but doesn't respect the "Block all third-party cookies" setting, or other reasons why the cookie might actually be blocked. To clarify, the actual cookie behavior is correct and matches the Tracking Protection dialog in the omnibox - all third-party cookies are blocked when the "Block all third-party cookies" setting is enabled.

I am following up on the Issues panel fix in https://issues.chromium.org/u/4/issues/325310946.

**Comment by wongbryan** [comment]
I’m not sure if that is true. If third party cookies are truly being blocked on that page then the app I am loading should NOT be able to authenticate itself with our servers. However, the plugin runs as expected, so I suspect it is using a validated session cookie from our server (which would be a third party cookie since this is in figma page).

**Comment by wanderview** [comment]
I still can't reproduce this.  Even with the editor link above the devtools warnings still go away when I block all 3P cookies and reload the figma tab.  Instead I see this in devtools:

![image](https://github.com/GoogleChromeLabs/privacy-sandbox-dev-support/assets/1256526/23749974-f0b2-4a71-babb-21041297e9dc)

**Comment by wongbryan** [comment]
@wanderview This is strange. I asked a coworker to block 3PC and opt into the 3PCD flag and test it. He is experiencing the same issue as I am where 3PC are allowed in Figma. Another coworker reported that 3PC are being blocked as they are supposed to be. I feel like this is probably a bug. Is there a way to investigate further?

**Comment by wanderview** [comment]
I think we would need to see your chrome://flags and chrome://settings/trackingProtection pages.  There is probably a setting somewhere that is inconsistent.

**Comment by wongbryan** [comment]
chrome://flags

<img width="835" alt="image" src="https://github.com/GoogleChromeLabs/privacy-sandbox-dev-support/assets/26299742/2dbc6dcc-fd06-437c-95a0-427b8273cc90">

chrome://settings/trackingProtection
<img width="770" alt="image" src="https://github.com/GoogleChromeLabs/privacy-sandbox-dev-support/assets/26299742/d7306af7-8be2-4977-8528-b2da8e30d515">

**Comment by johannhof** [comment]
Thanks @wongbryan, this seems to be an issue with the #test-third-party-cookies-phase flag. You can use the #tracking-protection-3pcd to the same effect but it shouldn't result in that message when blocking all third party cookies is turned on.

I'll also note that we might have to revisit showing this message for same-site origins (figma.com / www.figma.com), as that seems confusing, obviously *.figma.com is allowed to set cookies on that site, no matter the setting (except if you block first-party cookies).

**Comment by johannhof** [comment]
Ah @amaliev I think you already fixed the same-site issue with crrev.com/c/5296183, right?

**Comment by amaliev** [comment]
Yes, with that change we only show the message if a cookie is *allowed, because of* 3PCD mitigations. It will prevent firing the message for same-site cookie access as well.

**Comment by wongbryan** [comment]
Disabling #test-third-party-cookies-phaseout and enabling #tracking-protection-3pcd  works for me. thank you for the help!  I think I should flag here that enabling #test-third-party-cookies-phaseout not only causes the message shown above, but also was **mistakenly allowing 3PC on the webpage**. I was able to access features that required reading 3PC when the message was shown, so I dont think the bug was purely cosmetic. not sure if this useful but fyi
